mmengine/utils/version_utils.py

```diff
@@ -58,8 +58,9 @@ def _minimal_ext_cmd(cmd):
     env['LANGUAGE'] = 'C'
     env['LANG'] = 'C'
     env['LC_ALL'] = 'C'
-    out = subprocess.Popen(
-        cmd, stdout=subprocess.PIPE, env=env).communicate()[0]
+    out, err = subprocess.Popen(
```

Comment:
Should we throw a warning here by `print_log` ?

Reply:
Hi, I wasn't sure about warning, because if I set log level to INFO which is below WARNING, then I'll still end up having to see all the git error messages

Reply:
But the problem I see with that solution is it won't solve the problem of my logs being spammed with unuseful message from this function. Is there a way to do it so that I can see regular logging and hide these messages?

Reply:
Do you mean duplicated error messages like this?

![image](https://user-images.githubusercontent.com/57566630/201931417-815c1ea7-49b0-4db5-83a6-3b9c073d881f.png)

`print_log` will only output the rank:0 messages by default, do you mean that it will still be annoying to see the log like

```bash
{current_path} is not a git repositories
```

Reply:
Oh, yeah I mean even with just rank 0 printing I'd still see it at the end of each epoch while saving checkpoint like this:

```
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:30:53 - mmengine - INFO - Epoch(train) [1][100/782]  lr: 5.0000e-02  eta: 0:14:17  time: 0.0787  data_time: 0.0008  memory: 439  loss: 2.0481
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:31:01 - mmengine - INFO - Epoch(train) [1][200/782]  lr: 5.0000e-02  eta: 0:11:58  time: 0.0742  data_time: 0.0009  memory: 439  loss: 1.9125
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:31:09 - mmengine - INFO - Epoch(train) [1][300/782]  lr: 5.0000e-02  eta: 0:11:04  time: 0.0752  data_time: 0.0009  memory: 439  loss: 1.6958
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:31:17 - mmengine - INFO - Epoch(train) [1][400/782]  lr: 5.0000e-02  eta: 0:10:44  time: 0.0896  data_time: 0.0009  memory: 439  loss: 1.8454
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:31:25 - mmengine - INFO - Epoch(train) [1][500/782]  lr: 5.0000e-02  eta: 0:10:24  time: 0.0747  data_time: 0.0009  memory: 439  loss: 1.5863
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:31:32 - mmengine - INFO - Epoch(train) [1][600/782]  lr: 5.0000e-02  eta: 0:10:04  time: 0.0745  data_time: 0.0009  memory: 439  loss: 1.6130
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:31:40 - mmengine - INFO - Epoch(train) [1][700/782]  lr: 5.0000e-02  eta: 0:09:51  time: 0.0857  data_time: 0.0009  memory: 439  loss: 1.5064
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:31:47 - mmengine - INFO - Exp name: resnet34_8xb16_cifar10_20221110_183027
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:31:47 - mmengine - INFO - Saving checkpoint at 1 epochs
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stderr>:fatal: not a git repository (or any of the parent directories): .git
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:31:55 - mmengine - INFO - Epoch(train) [2][100/782]  lr: 5.0000e-02  eta: 0:09:25  time: 0.0725  data_time: 0.0009  memory: 439  loss: 1.4329
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:32:03 - mmengine - INFO - Epoch(train) [2][200/782]  lr: 5.0000e-02  eta: 0:09:13  time: 0.0762  data_time: 0.0009  memory: 439  loss: 1.3959
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:32:04 - mmengine - INFO - Exp name: resnet34_8xb16_cifar10_20221110_183027
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:32:10 - mmengine - INFO - Epoch(train) [2][300/782]  lr: 5.0000e-02  eta: 0:08:59  time: 0.0618  data_time: 0.0009  memory: 439  loss: 1.0014
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:32:17 - mmengine - INFO - Epoch(train) [2][400/782]  lr: 5.0000e-02  eta: 0:08:49  time: 0.0750  data_time: 0.0009  memory: 439  loss: 1.0745
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:32:25 - mmengine - INFO - Epoch(train) [2][500/782]  lr: 5.0000e-02  eta: 0:08:40  time: 0.0853  data_time: 0.0010  memory: 439  loss: 1.1572
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:32:33 - mmengine - INFO - Epoch(train) [2][600/782]  lr: 5.0000e-02  eta: 0:08:32  time: 0.0783  data_time: 0.0009  memory: 439  loss: 0.8799
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:32:41 - mmengine - INFO - Epoch(train) [2][700/782]  lr: 5.0000e-02  eta: 0:08:23  time: 0.0792  data_time: 0.0009  memory: 439  loss: 0.9982
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:32:47 - mmengine - INFO - Exp name: resnet34_8xb16_cifar10_20221110_183027
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:32:47 - mmengine - INFO - Saving checkpoint at 2 epochs
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stderr>:fatal: not a git repository (or any of the parent directories): .git
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:32:51 - mmengine - INFO - Epoch(val) [2][100/157]    eta: 0:00:01  time: 0.0274  data_time: 0.0006  memory: 439
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:32:56 - mmengine - INFO - Epoch(val) [2][157/157]  accuracy/top1: 65.1200
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stderr>:fatal: not a git repository (or any of the parent directories): .git
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:32:56 - mmengine - INFO - The best checkpoint with 65.1200 accuracy/top1 at 2 epoch is saved to best_accuracy/top1_epoch_2.pth.
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:33:04 - mmengine - INFO - Epoch(train) [3][100/782]  lr: 5.0000e-02  eta: 0:08:08  time: 0.0768  data_time: 0.0009  memory: 439  loss: 1.0032
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:33:12 - mmengine - INFO - Epoch(train) [3][200/782]  lr: 5.0000e-02  eta: 0:07:59  time: 0.0696  data_time: 0.0009  memory: 439  loss: 0.8434
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:33:20 - mmengine - INFO - Epoch(train) [3][300/782]  lr: 5.0000e-02  eta: 0:07:51  time: 0.0875  data_time: 0.0009  memory: 439  loss: 1.0312
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:33:28 - mmengine - INFO - Epoch(train) [3][400/782]  lr: 5.0000e-02  eta: 0:07:43  time: 0.0760  data_time: 0.0009  memory: 439  loss: 1.0394
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:33:31 - mmengine - INFO - Exp name: resnet34_8xb16_cifar10_20221110_183027
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:33:36 - mmengine - INFO - Epoch(train) [3][500/782]  lr: 5.0000e-02  eta: 0:07:35  time: 0.0731  data_time: 0.0009  memory: 439  loss: 0.8294
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:33:43 - mmengine - INFO - Epoch(train) [3][600/782]  lr: 5.0000e-02  eta: 0:07:26  time: 0.0750  data_time: 0.0009  memory: 439  loss: 0.8393
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:33:51 - mmengine - INFO - Epoch(train) [3][700/782]  lr: 5.0000e-02  eta: 0:07:18  time: 0.0747  data_time: 0.0009  memory: 439  loss: 0.8261
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:33:57 - mmengine - INFO - Exp name: resnet34_8xb16_cifar10_20221110_183027
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:33:57 - mmengine - INFO - Saving checkpoint at 3 epochs
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stderr>:fatal: not a git repository (or any of the parent directories): .git
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:34:06 - mmengine - INFO - Epoch(train) [4][100/782]  lr: 5.0000e-02  eta: 0:07:03  time: 0.0778  data_time: 0.0010  memory: 439  loss: 0.8342
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:34:14 - mmengine - INFO - Epoch(train) [4][200/782]  lr: 5.0000e-02  eta: 0:06:55  time: 0.0772  data_time: 0.0010  memory: 439  loss: 0.6367
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:34:21 - mmengine - INFO - Epoch(train) [4][300/782]  lr: 5.0000e-02  eta: 0:06:46  time: 0.0630  data_time: 0.0009  memory: 439  loss: 0.7741
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:34:27 - mmengine - INFO - Epoch(train) [4][400/782]  lr: 5.0000e-02  eta: 0:06:35  time: 0.0612  data_time: 0.0008  memory: 439  loss: 0.7217
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:34:33 - mmengine - INFO - Epoch(train) [4][500/782]  lr: 5.0000e-02  eta: 0:06:24  time: 0.0621  data_time: 0.0008  memory: 439  loss: 0.5821
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:34:39 - mmengine - INFO - Epoch(train) [4][600/782]  lr: 5.0000e-02  eta: 0:06:14  time: 0.0596  data_time: 0.0008  memory: 439  loss: 0.6138
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:34:43 - mmengine - INFO - Exp name: resnet34_8xb16_cifar10_20221110_183027
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:34:45 - mmengine - INFO - Epoch(train) [4][700/782]  lr: 5.0000e-02  eta: 0:06:03  time: 0.0589  data_time: 0.0008  memory: 439  loss: 0.6686
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:34:51 - mmengine - INFO - Exp name: resnet34_8xb16_cifar10_20221110_183027
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:34:51 - mmengine - INFO - Saving checkpoint at 4 epochs
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stderr>:fatal: not a git repository (or any of the parent directories): .git
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:34:53 - mmengine - INFO - Epoch(val) [4][100/157]    eta: 0:00:01  time: 0.0202  data_time: 0.0005  memory: 439
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:34:57 - mmengine - INFO - Epoch(val) [4][157/157]  accuracy/top1: 75.7400
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:34:57 - mmengine - INFO - The previous best checkpoint /opt/ml/checkpoints/best_accuracy/top1_epoch_2.pth is removed
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stderr>:fatal: not a git repository (or any of the parent directories): .git
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:34:58 - mmengine - INFO - The best checkpoint with 75.7400 accuracy/top1 at 4 epoch is saved to best_accuracy/top1_epoch_4.pth.
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:35:04 - mmengine - INFO - Epoch(train) [5][100/782]  lr: 5.0000e-02  eta: 0:05:46  time: 0.0652  data_time: 0.0008  memory: 439  loss: 0.4876
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:35:09 - mmengine - INFO - Epoch(train) [5][200/782]  lr: 5.0000e-02  eta: 0:05:36  time: 0.0588  data_time: 0.0008  memory: 439  loss: 0.5643
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:35:15 - mmengine - INFO - Epoch(train) [5][300/782]  lr: 5.0000e-02  eta: 0:05:26  time: 0.0554  data_time: 0.0007  memory: 439  loss: 0.5708
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:35:21 - mmengine - INFO - Epoch(train) [5][400/782]  lr: 5.0000e-02  eta: 0:05:17  time: 0.0580  data_time: 0.0007  memory: 439  loss: 0.7676
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:35:27 - mmengine - INFO - Epoch(train) [5][500/782]  lr: 5.0000e-02  eta: 0:05:08  time: 0.0696  data_time: 0.0007  memory: 439  loss: 0.7844
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:35:33 - mmengine - INFO - Epoch(train) [5][600/782]  lr: 5.0000e-02  eta: 0:04:59  time: 0.0580  data_time: 0.0008  memory: 439  loss: 0.5548
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:35:39 - mmengine - INFO - Epoch(train) [5][700/782]  lr: 5.0000e-02  eta: 0:04:50  time: 0.0677  data_time: 0.0009  memory: 439  loss: 0.4288
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:35:43 - mmengine - INFO - Exp name: resnet34_8xb16_cifar10_20221110_183027
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:35:43 - mmengine - INFO - Saving checkpoint at 5 epochs
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stderr>:fatal: not a git repository (or any of the parent directories): .git
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:35:49 - mmengine - INFO - Exp name: resnet34_8xb16_cifar10_20221110_183027
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:35:50 - mmengine - INFO - Epoch(train) [6][100/782]  lr: 5.0000e-02  eta: 0:04:34  time: 0.0584  data_time: 0.0008  memory: 439  loss: 0.5210
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:35:55 - mmengine - INFO - Epoch(train) [6][200/782]  lr: 5.0000e-02  eta: 0:04:25  time: 0.0556  data_time: 0.0008  memory: 439  loss: 0.6091
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:36:01 - mmengine - INFO - Epoch(train) [6][300/782]  lr: 5.0000e-02  eta: 0:04:17  time: 0.0608  data_time: 0.0008  memory: 439  loss: 0.4485
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:36:07 - mmengine - INFO - Epoch(train) [6][400/782]  lr: 5.0000e-02  eta: 0:04:09  time: 0.0565  data_time: 0.0007  memory: 439  loss: 0.4674
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:36:13 - mmengine - INFO - Epoch(train) [6][500/782]  lr: 5.0000e-02  eta: 0:04:00  time: 0.0573  data_time: 0.0008  memory: 439  loss: 0.4487
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:36:18 - mmengine - INFO - Epoch(train) [6][600/782]  lr: 5.0000e-02  eta: 0:03:52  time: 0.0558  data_time: 0.0007  memory: 439  loss: 0.4053
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:36:24 - mmengine - INFO - Epoch(train) [6][700/782]  lr: 5.0000e-02  eta: 0:03:44  time: 0.0596  data_time: 0.0008  memory: 439  loss: 0.4595
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:36:29 - mmengine - INFO - Exp name: resnet34_8xb16_cifar10_20221110_183027
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:36:29 - mmengine - INFO - Saving checkpoint at 6 epochs
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stderr>:fatal: not a git repository (or any of the parent directories): .git
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:36:32 - mmengine - INFO - Epoch(val) [6][100/157]    eta: 0:00:01  time: 0.0193  data_time: 0.0006  memory: 439
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:36:35 - mmengine - INFO - Epoch(val) [6][157/157]  accuracy/top1: 81.9000
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:36:35 - mmengine - INFO - The previous best checkpoint /opt/ml/checkpoints/best_accuracy/top1_epoch_4.pth is removed
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stderr>:fatal: not a git repository (or any of the parent directories): .git
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:36:36 - mmengine - INFO - The best checkpoint with 81.9000 accuracy/top1 at 6 epoch is saved to best_accuracy/top1_epoch_6.pth.
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:36:41 - mmengine - INFO - Epoch(train) [7][100/782]  lr: 5.0000e-02  eta: 0:03:30  time: 0.0570  data_time: 0.0007  memory: 439  loss: 0.3789
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:36:47 - mmengine - INFO - Epoch(train) [7][200/782]  lr: 5.0000e-02  eta: 0:03:23  time: 0.0575  data_time: 0.0007  memory: 439  loss: 0.6134
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:36:53 - mmengine - INFO - Epoch(train) [7][300/782]  lr: 5.0000e-02  eta: 0:03:15  time: 0.0634  data_time: 0.0008  memory: 439  loss: 0.4193
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:36:53 - mmengine - INFO - Exp name: resnet34_8xb16_cifar10_20221110_183027
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:36:59 - mmengine - INFO - Epoch(train) [7][400/782]  lr: 5.0000e-02  eta: 0:03:07  time: 0.0578  data_time: 0.0008  memory: 439  loss: 0.4403
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:37:05 - mmengine - INFO - Epoch(train) [7][500/782]  lr: 5.0000e-02  eta: 0:03:00  time: 0.0579  data_time: 0.0008  memory: 439  loss: 0.4681
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:37:11 - mmengine - INFO - Epoch(train) [7][600/782]  lr: 5.0000e-02  eta: 0:02:53  time: 0.0608  data_time: 0.0008  memory: 439  loss: 0.4626
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:37:16 - mmengine - INFO - Epoch(train) [7][700/782]  lr: 5.0000e-02  eta: 0:02:45  time: 0.0561  data_time: 0.0008  memory: 439  loss: 0.4237
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:37:21 - mmengine - INFO - Exp name: resnet34_8xb16_cifar10_20221110_183027
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:37:21 - mmengine - INFO - Saving checkpoint at 7 epochs
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stderr>:fatal: not a git repository (or any of the parent directories): .git
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:37:28 - mmengine - INFO - Epoch(train) [8][100/782]  lr: 5.0000e-02  eta: 0:02:32  time: 0.0594  data_time: 0.0008  memory: 439  loss: 0.3859
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:37:33 - mmengine - INFO - Epoch(train) [8][200/782]  lr: 5.0000e-02  eta: 0:02:25  time: 0.0568  data_time: 0.0007  memory: 439  loss: 0.3482
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:37:39 - mmengine - INFO - Epoch(train) [8][300/782]  lr: 5.0000e-02  eta: 0:02:18  time: 0.0574  data_time: 0.0007  memory: 439  loss: 0.5458
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:37:45 - mmengine - INFO - Epoch(train) [8][400/782]  lr: 5.0000e-02  eta: 0:02:11  time: 0.0581  data_time: 0.0008  memory: 439  loss: 0.5149
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:37:50 - mmengine - INFO - Epoch(train) [8][500/782]  lr: 5.0000e-02  eta: 0:02:04  time: 0.0615  data_time: 0.0009  memory: 439  loss: 0.3969
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:37:52 - mmengine - INFO - Exp name: resnet34_8xb16_cifar10_20221110_183027
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:37:56 - mmengine - INFO - Epoch(train) [8][600/782]  lr: 5.0000e-02  eta: 0:01:57  time: 0.0571  data_time: 0.0008  memory: 439  loss: 0.4906
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:38:02 - mmengine - INFO - Epoch(train) [8][700/782]  lr: 5.0000e-02  eta: 0:01:50  time: 0.0575  data_time: 0.0008  memory: 439  loss: 0.3588
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:38:07 - mmengine - INFO - Exp name: resnet34_8xb16_cifar10_20221110_183027
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:38:07 - mmengine - INFO - Saving checkpoint at 8 epochs
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stderr>:fatal: not a git repository (or any of the parent directories): .git
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:38:10 - mmengine - INFO - Epoch(val) [8][100/157]    eta: 0:00:01  time: 0.0193  data_time: 0.0005  memory: 439
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:38:13 - mmengine - INFO - Epoch(val) [8][157/157]  accuracy/top1: 84.0600
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:38:13 - mmengine - INFO - The previous best checkpoint /opt/ml/checkpoints/best_accuracy/top1_epoch_6.pth is removed
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stderr>:fatal: not a git repository (or any of the parent directories): .git
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:38:14 - mmengine - INFO - The best checkpoint with 84.0600 accuracy/top1 at 8 epoch is saved to best_accuracy/top1_epoch_8.pth.
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:38:19 - mmengine - INFO - Epoch(train) [9][100/782]  lr: 5.0000e-02  eta: 0:01:37  time: 0.0608  data_time: 0.0009  memory: 439  loss: 0.3629
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:38:25 - mmengine - INFO - Epoch(train) [9][200/782]  lr: 5.0000e-02  eta: 0:01:30  time: 0.0600  data_time: 0.0009  memory: 439  loss: 0.3150
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:38:31 - mmengine - INFO - Epoch(train) [9][300/782]  lr: 5.0000e-02  eta: 0:01:23  time: 0.0662  data_time: 0.0009  memory: 439  loss: 0.3463
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:38:37 - mmengine - INFO - Epoch(train) [9][400/782]  lr: 5.0000e-02  eta: 0:01:17  time: 0.0605  data_time: 0.0007  memory: 439  loss: 0.4144
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:38:43 - mmengine - INFO - Epoch(train) [9][500/782]  lr: 5.0000e-02  eta: 0:01:10  time: 0.0561  data_time: 0.0008  memory: 439  loss: 0.4026
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:38:48 - mmengine - INFO - Epoch(train) [9][600/782]  lr: 5.0000e-02  eta: 0:01:03  time: 0.0572  data_time: 0.0007  memory: 439  loss: 0.4822
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:38:54 - mmengine - INFO - Epoch(train) [9][700/782]  lr: 5.0000e-02  eta: 0:00:56  time: 0.0600  data_time: 0.0008  memory: 439  loss: 0.4339
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:38:57 - mmengine - INFO - Exp name: resnet34_8xb16_cifar10_20221110_183027
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:38:59 - mmengine - INFO - Exp name: resnet34_8xb16_cifar10_20221110_183027
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:38:59 - mmengine - INFO - Saving checkpoint at 9 epochs
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stderr>:fatal: not a git repository (or any of the parent directories): .git
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:39:05 - mmengine - INFO - Epoch(train) [10][100/782]  lr: 5.0000e-02  eta: 0:00:44  time: 0.0557  data_time: 0.0008  memory: 439  loss: 0.4057
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:39:11 - mmengine - INFO - Epoch(train) [10][200/782]  lr: 5.0000e-02  eta: 0:00:38  time: 0.0605  data_time: 0.0009  memory: 439  loss: 0.3223
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:39:17 - mmengine - INFO - Epoch(train) [10][300/782]  lr: 5.0000e-02  eta: 0:00:31  time: 0.0556  data_time: 0.0007  memory: 439  loss: 0.4687
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:39:23 - mmengine - INFO - Epoch(train) [10][400/782]  lr: 5.0000e-02  eta: 0:00:24  time: 0.0562  data_time: 0.0007  memory: 439  loss: 0.4386
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:39:28 - mmengine - INFO - Epoch(train) [10][500/782]  lr: 5.0000e-02  eta: 0:00:18  time: 0.0552  data_time: 0.0008  memory: 439  loss: 0.3845
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:39:34 - mmengine - INFO - Epoch(train) [10][600/782]  lr: 5.0000e-02  eta: 0:00:11  time: 0.0569  data_time: 0.0008  memory: 439  loss: 0.3783
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:39:40 - mmengine - INFO - Epoch(train) [10][700/782]  lr: 5.0000e-02  eta: 0:00:05  time: 0.0559  data_time: 0.0008  memory: 439  loss: 0.2946
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:39:44 - mmengine - INFO - Exp name: resnet34_8xb16_cifar10_20221110_183027
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:39:44 - mmengine - INFO - Saving checkpoint at 10 epochs
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stderr>:fatal: not a git repository (or any of the parent directories): .git
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:39:47 - mmengine - INFO - Epoch(val) [10][100/157]    eta: 0:00:01  time: 0.0183  data_time: 0.0006  memory: 439
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:39:51 - mmengine - INFO - Epoch(val) [10][157/157]  accuracy/top1: 85.7400
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:39:51 - mmengine - INFO - The previous best checkpoint /opt/ml/checkpoints/best_accuracy/top1_epoch_8.pth is removed
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stderr>:fatal: not a git repository (or any of the parent directories): .git
ilgjo6i41z-algo-1-4v75j | [1,mpirank:0,algo-1]<stdout>:11/10 18:39:51 - mmengine - INFO - The best checkpoint with 85.7400 accuracy/top1 at 10 epoch is saved to best_accuracy/top1_epoch_10.pth.
```

Reply:
Alright, got it, it seems the saved checkpoints requires the git hash each time, and I think the current modification makes sense to me :smiley: .

Reply:
Thanks, appreciate it!